portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -47,7 +47,8 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       //TODO: use GSON deserialization
       String name = request.getParameter("name");
       String content = request.getParameter("content");
-      Comment comment = new Comment(0, name, content);
+      Comment comment = new Comment(0, name, content); //Since the ID will only be assigned by the Datastore, I set it to 0. 
```

Comment:
Instead create a constructor without the id, just passing `name` and `content`

Reply:
Alternatively a builder pattern could be used to create an instance. See e.g. https://dzone.com/articles/design-patterns-the-builder-pattern

Reply:
Ohh this is interesting. I think here it is really a good solution, but I have a theoretical question: with the builder pattern, how can you ensure, that you have set every necessary parameters? (The constructor would give you an error during compilation if it missed something.)

Reply:
With the builder, after setting every field, you call `build()` to get the final object - and that is the place you can do a validation.

Reply:
But that will only give me an exception in runtime, right?

Reply:
Yes. That's why builders I've seen were used mostly for large objects that had most of the fields optional.
In you case, a second constructor is the best option.